daemon/state.go

```diff
@@ -134,6 +133,11 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 	for _, ep := range state.restored {
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
 			ep.Mutex.Lock()
+
+			// Insert into endpoint manager so it can be regenerated when calls to
+			// TriggerPolicyUpdates() are made.
+			endpointmanager.Insert(ep)
+
 			scopedLog := log.WithField(logfields.EndpointID, ep.ID)
 
 			state := ep.GetStateLocked()
```

Comment:
Given this change, you can remove this code do detect and abort on disconnected state again as there is no way that an endpoint could have been disconnected in the meantime as it was not accessible in any way.